From 9333f3f79a253fcdd792bee9f5bd7cfb856a44d9 Mon Sep 17 00:00:00 2001
From: Peter Jung <admin@ptr1337.dev>
Date: Thu, 30 Jan 2025 18:40:18 +0100
Subject: [PATCH 1/3] Enable atomic kernel modesetting by default

This is required for proper functionality under Wayland. fbdev has been default enabled since 570 so that
hunk can be removed from this patch.

Signed-off-by: Eric Naim <dnaim@cachyos.org>
Signed-off-by: Peter Jung <admin@ptr1337.dev>
---
 nvidia-drm/nvidia-drm-linux.c        | 2 +-
 nvidia-drm/nvidia-drm-os-interface.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/nvidia-drm/nvidia-drm-linux.c b/nvidia-drm/nvidia-drm-linux.c
index 3cb1815d..209cb469 100644
--- a/nvidia-drm/nvidia-drm-linux.c
+++ b/nvidia-drm/nvidia-drm-linux.c
@@ -31,7 +31,7 @@
 
 MODULE_PARM_DESC(
     modeset,
-    "Enable atomic kernel modesetting (1 = enable, 0 = disable (default))");
+    "Enable atomic kernel modesetting (1 = enable (default), 0 = disable)");
 module_param_named(modeset, nv_drm_modeset_module_param, bool, 0400);
 
 #if defined(NV_DRM_FBDEV_AVAILABLE)
diff --git a/nvidia-drm/nvidia-drm-os-interface.c b/nvidia-drm/nvidia-drm-os-interface.c
index 7617476d..f22afd77 100644
--- a/nvidia-drm/nvidia-drm-os-interface.c
+++ b/nvidia-drm/nvidia-drm-os-interface.c
@@ -41,7 +41,7 @@
 #include <drm/drmP.h>
 #endif
 
-bool nv_drm_modeset_module_param = false;
+bool nv_drm_modeset_module_param = true;
 bool nv_drm_fbdev_module_param = true;
 
 void *nv_drm_calloc(size_t nmemb, size_t size)
-- 
2.48.1

